swap_meet/vendor.py

```diff
@@ -1,2 +1,80 @@
 class Vendor:
-    pass
\ No newline at end of file
+        def __init__(self, inventory = None):
+            if not inventory:
+                inventory = []
+            self.inventory = inventory
+
+        def add(self, item):
+            if not item in self.inventory:
+                self.inventory.append(item)
+                return item
+
+        def remove(self, item):
+            if item in self.inventory:
+                self.inventory.remove(item)
+                return item
+            else:
+                return False
+        
+        def get_by_category(self, category):
+            item_list = []
+            for item in self.inventory:
+                if category == item.category:
+                    item_list.append(item)
+            return item_list
+
+        def swap_items(self, vendor, my_item, their_item):
+            if my_item not in self.inventory or their_item not in vendor.inventory:
+                return False
+            self.inventory.remove(my_item)
+            vendor.remove(their_item)
+            self.inventory.append(their_item)
+            vendor.inventory.append(my_item)
+
+            return True
+        def swap_first_item(self, vendor):
+            if len(self.inventory) == 0 or len(vendor.inventory) == 0:
+                return False
+            # Get first item from inventory
+            first_item_inventory = self.inventory[0]
+            # Get first item from their inventory
+            first_item_vendor = vendor.inventory[0]
+            # Remove first item from inventory 
+            self.inventory.remove(first_item_inventory)
+            vendor.inventory.remove(first_item_vendor)
+            # "Update" first item on their inventory 
+            vendor.inventory.insert(0, first_item_inventory)
+            self.inventory.insert(0,first_item_vendor )
```

Comment:
Consider using your `swap_items` method and pass in f`irst_item_inventory` and `first_item_vendor` since that logic to remove and add already exists. 

I see that you use insert to swap the first items and put them in the 0th position. Since the requirements doesn't require the first item to be swapped into the first position, I think it's fair game to append it to the end of the inventory list.